File: fixtures/showcase.ts
type ThemeMode = "light" | "dark";

interface ThemePreference {
  mode: ThemeMode;
  contrast: number;
}

export function describeTheme(preference: ThemePreference): string {
  const readableContrast = Math.max(preference.contrast, 4.5);
  return `${preference.mode} theme at ${readableContrast}:1`;
}

File: fixtures/showcase.cpp
#include <algorithm>
#include <string>

namespace everforest {
struct ThemePreference {
  std::string mode;
  double contrast;
};

double readableContrast(const ThemePreference &preference) {
  return std::max(preference.contrast, 4.5);
}
}  // namespace everforest
